src/openfl/_internal/renderer/canvas/CanvasTilemap.hx

```diff
@@ -84,38 +83,14 @@ class CanvasTilemap {
 			}
 			
 			bitmapData = tileset.bitmapData;
-			if (bitmapData == null || !bitmapData.__prepareImage()) continue;
-			
-			if (bitmapData != cacheBitmapData) {
-				
-				if (bitmapData.image.buffer.__srcImage == null) {
-					
-					ImageCanvasUtil.convertToCanvas (bitmapData.image);
-					
-				}
-				
-				source = bitmapData.image.src;
-				cacheBitmapData = bitmapData;
-				
-			}
+			if (bitmapData == null || !bitmapData.__canBeDrawnToCanvas()) continue;
 			
 			context.globalAlpha = tilemap.__worldAlpha * alpha;
 			
 			tileTransform = tile.matrix;
 			tileTransform.concat (transform);
-			var pixelRatio = renderSession.pixelRatio;
-			
-			if (roundPixels) {
-				
-				context.setTransform (tileTransform.a * pixelRatio, tileTransform.b, tileTransform.c, tileTransform.d * pixelRatio, Math.round (tileTransform.tx * pixelRatio), Math.round (tileTransform.ty * pixelRatio));
-				
-			} else {
-				
-				context.setTransform (tileTransform.a * pixelRatio, tileTransform.b, tileTransform.c, tileTransform.d * pixelRatio, tileTransform.tx * pixelRatio, tileTransform.ty * pixelRatio);
-				
-			}
 			
-			context.drawImage (source, tileRect.x, tileRect.y, tileRect.width, tileRect.height, 0, 0, tileRect.width, tileRect.height);
+			bitmapData.__drawToCanvas (context, tileTransform, roundPixels, renderSession.pixelRatio, tileRect, false);
```

Comment:
It is actually not used here, but haven't we at some point added `__snapToPixel` to DisplayObject to use it for Tilemap as well? Can you quickly check what the GLTilemap does?

Reply:
Yes, `roundPixels` is a local var initialized as `var roundPixels = renderSession.roundPixels || tilemap.__snapToPixel ();`

Reply:
(GLTilemap also uses `__snapToPixel`)

Reply:
Ah true, I see, it's local and at the top. Sorry ;)